lib/lever/client.rb

```diff
@@ -72,20 +108,41 @@ def post_resource(path, body)
       response.parsed_response
     end
 
+    def with_retries
+      return yield if using_with_retries
+
+      begin
+        # Eventually we want to have lower-level methods like #get_resource implement retries automatically
+        # So let's disallow nested `with_retries` blocks just in case we add it there but forget to remove it from
+        #   higher-level methods
+        self.using_with_retries = true
+
+        Retriable.retriable(on: [Lever::TooManyRequestsError, Lever::ServerError, Lever::ServiceUnavailableError]) do
```

Comment:
This is slightly different from the greenhouse gem b/c we're not reacting to _any_ 5xx error (only 500 and 503), but I think that's fine per https://hire.sandbox.lever.co/developer/documentation#responses-and-errors